### PR TITLE
merges engineer with mechanic

### PR DESCRIPTION
**Pull request by Adharainspace**
[ENHANCEMENT]

## About the PR 
disables mechanic as a job you can join / latejoin / set preferences for. adds mechanic access to engineers by default, and moves mechanic objectives to engineers as well. moves mechanic traitor items to engineer. increases engineer slots by 1 across maps. leaves some admin stuff re: mechanics in, because im afraid to touch that tbh. (might be for debugging mechanics system, and ones abt lifting antag bans??)
also i tested it and it seems like it works.

## Why's this needed? 
this was discussed in discord a bit, and it seems like a good idea. the biggest comparison i can draw for this is between chemist and scientist, before the two were merged. i think that mechanics dont really have a lot to do, and one way to make the job less aimless and better with the station is to merge it with engineering. a lot of people have trouble getting mechanics to work with engineers, like, an engineer will patch the breach in cloning and refill the air, but then you end up waiting forever for the mechanic and end up just doing it yourself. this also solves the issue of engineers not having much to do after starting the engine, because then they can work on mech stuff and other station improvements. and engineers can actually set up ptl teleporters now without having to wait for mechanics to stop loafing or building another suicide booth. 
of course im open to other suggestions too.

## Changelog

```
(u)Adhara:
(*)Removed the mechanic job and merged its responsibilities with engineer.
```

### CODE REVIEW

**Comment by MarkNais** [comment]
Seems reasonable.
Also, clever code change on the job-limit.

**Comment by Adharainspace** [comment]
removes commented out mechanics stuff, apparently its easier to revert than uncomment, so there

**Comment by UrsulaMejor** [comment]
I like the idea, tbh. The two jobs serve nearly identical purposes, but given that Mechanics access/tools is often needed to complete the job of Engineering in fixing up the station, it seems natural to merge them

**Comment by zjdtmkhzt** [comment]
Giving engineers more things that they can do after they finish setting up the engine seems like a good thing to me.
The only thing I am worried about is all of engineering spawning in because they wanna do stupid mechanics things and none of them wanting to set up the engine. Unlike Scientists, Engineers have an actual job to do. 🤔

**Comment by TTerc** [comment]
Yes, I like the idea of converting the engineer job to make it more versatile, as I'm a fan of versatility of certain jobs (scientist, for example) in general. I won't have to break into engi storage to grab metal sheets in case of a breach anymore, and I will be able to take care of the engine while also being able to mess with mechanics equipment.
Sounds good to me

**Comment by glassofmilk** [comment]
I don't like it. I am probably biased as someone who almost always plays Mechanic, but I really like having a separate identity from Engineering. Also, the Mechanic jumpsuits are much nicer. Please leave in the Mechanic gear lockers, at least. If you want to do Mechanic stuff, being able to get a Mechanic PDA with a Systems Analysis cart and T-Ray module makes things WAY easier. You may want to add the "engineering" PDA group to the Mechanic PDAs, though.

File: code/procs/access.dm [hunk 1]
```diff
@@ -298,7 +298,8 @@
 						access_eva, access_engineering, access_engineering_storage, access_engineering_eva, access_engineering_atmos)
 		if("Engineer")
 			return list(access_engineering,access_maint_tunnels,access_external_airlocks,
-						access_engineering_storage,access_engineering_atmos,access_engineering_engine,access_engineering_power)
+						access_engineering_storage, access_engineering_atmos, access_engineering_engine,
+						access_engineering_power, access_engineering_mechanic)
```
Comment (glassofmilk) [inline code/procs/access.dm:302]: Engineers will also need access_tech_storage

**Comment by AwkwardDryad** [comment]
When I first started playing space station, I queued up as an engineer thinking it would be something involving managing station power and building neat things, but those things have been separated into mechanic and engineer. Seeing them merged makes me very happy for not only that reason, but because after the engine is set up, engineers really don't have much to do. The station repair is mainly carried out by mechanics anyway, despite it being under the job description of engineers. In short, I think this is a wonderful idea :heart:

Also to respond to concerns about the engine potentially not being set up because of engineers wanting to do mechanics.
I feel like naturally, at least one person will feel obligated to set it up, and if not, there's always borgs. I think this is an issue that should be addressed if it ever comes up.

**Comment by ryebek** [comment]
I'm not really a fan, due to the jobs having a very different personality from one another. Mechanics is essentially a blue staff assistant, whose sole purpose is to break into places, and mass produce things they shouldn't be, while engineers are expected to actually do something, like keep the station powered, and stop it falling apart. I've never seen an engineer do a mechanics job, and I've barely ever seen a mechanic do their *own* job, and they don't feel anything alike in terms of how they're played. 

Both are fairly popular jobs, and merging them into one will just make the slots fill up quicker. With scientist/chemist being one job, the job slots are very regularly full, mainly because of people who want to do chemistry. I feel like if this gets merged, the same thing will happen just because people want to work on the engine, because the engineer slots are full fairly often already, without people joining to do mechanic stuff instead. By having them separate, the people who want to do engineer stuff doesn't stop the people who want to do mechanic stuff from doing so, or vice versa, because you can choose the job you want to do.

Finally, mechanics have a large capacity for fucking around, and having three potential culprits is much easier than eight, when they inevitably do something that pisses off security, which while less important than the other points, is still a valid point. Its much easier to identify which of the three mechanics has set up a PDA message spam script than it is to figure out which of the eight engineers has been doing nerd stuff, and which one of those few engineers did it. Its already hard enough. Big fat no from me.

**Comment by AwkwardDryad** [comment]
> Finally, mechanics have a large capacity for fucking around, and having three potential culprits is much easier than eight, when they inevitably do something that pisses off security, which while less important than the other points, is still a valid point. Its much easier to identify which of the three mechanics has set up a PDA message spam script than it is to figure out which of the eight engineers has been doing nerd stuff, and which one of those few engineers did it. Its already hard enough. Big fat no from me.

Eek that is a good point...

File: code/procs/access.dm (hunk 1 above)
Comment (Adharainspace) [reply code/procs/access.dm:302]: good catch!

**Comment by Adharainspace** [comment]
> Giving engineers more things that they can do after they finish setting up the engine seems like a good thing to me.
The only thing I am worried about is all of engineering spawning in because they wanna do stupid mechanics things and none of them wanting to set up the engine. Unlike Scientists, Engineers have an actual job to do. 🤔

im hoping that thisll be resolved by people just talking, enough people join engineer already to do the engine that there should be room for people to screw around roundstart

> I don't like it. I am probably biased as someone who almost always plays Mechanic, but I really like having a separate identity from Engineering. Also, the Mechanic jumpsuits are much nicer. Please leave in the Mechanic gear lockers, at least. If you want to do Mechanic stuff, being able to get a Mechanic PDA with a Systems Analysis cart and T-Ray module makes things WAY easier. You may want to add the "engineering" PDA group to the Mechanic PDAs, though.

im not removing mechanics gear lockers, ill add the engineering group to mechanics pdas too, but idk, i feel like it defeats the purpose if someone joins as engineer, just to suit up in mechanic outfit with mechanic pda to become a mechanic in everything but job title. maybe a merged uniform for engies that looks good? and mech pda carts but no mech pda in the locker? as is the lockers contain everything they had before, i can amend that to be just carts, or contain new/old outfits, or something. ideas welcome. 

> I'm not really a fan, due to the jobs having a very different personality from one another. Mechanics is essentially a blue staff assistant, whose sole purpose is to break into places, and mass produce things they shouldn't be, while engineers are expected to actually do something, like keep the station powered, and stop it falling apart. I've never seen an engineer do a mechanics job, and I've barely ever seen a mechanic do their _own_ job, and they don't feel anything alike in terms of how they're played.
> 
> Both are fairly popular jobs, and merging them into one will just make the slots fill up quicker. With scientist/chemist being one job, the job slots are very regularly full, mainly because of people who want to do chemistry. I feel like if this gets merged, the same thing will happen just because people want to work on the engine, because the engineer slots are full fairly often already, without people joining to do mechanic stuff instead. By having them separate, the people who want to do engineer stuff doesn't stop the people who want to do mechanic stuff from doing so, or vice versa, because you can choose the job you want to do.
> 
> Finally, mechanics have a large capacity for fucking around, and having three potential culprits is much easier than eight, when they inevitably do something that pisses off security, which while less important than the other points, is still a valid point. Its much easier to identify which of the three mechanics has set up a PDA message spam script than it is to figure out which of the eight engineers has been doing nerd stuff, and which one of those few engineers did it. Its already hard enough. Big fat no from me.

this makes some sense to me, because it seems like they have different roles. and idk, i dont wanna kill mechanic culture, but id rather have assistants be blue assistants who break into places and make things they shouldnt, instead of having that be the purpose of a job who's main responsibility is to restore cloning and medbay machines when they break down. this doesnt make it any harder for assistants to break into mech and steal insuls and go to town or anything. 

also, the amount of slots for engineers is malleable, im up to go and increase it if it means more people will join engineer, so that no job slots are effectively lost. i already addressed the engine stuff too. and i think that 8 players for a job is plenty. like, theres only 1 chef slot, sometimes you dont get the job you want because usually 60 other people might want that job too, you know? 6-8 engineer slots is a lot, and gives people space to set up  the engine, or just loaf around (pun intended 🕶️)

i think the point i agree with the most is your last one, mechanics currently have a really large potential to be a huge annoyance to the station, and blending mechanics / engineers together would muddy that a lot. but i still think you could use detective skills ("hey who was working on the engine with you", scanning for prints / glove fibers, whatever else an investigator does) to figure stuff out. just like if a scientist chembombed medbay, it would be a larger pool of suspects but you could narrow it down

**Comment by Flaborized** [comment]
I like this, I think it'd make playing engineer more fun and make repairing the station or building improvements a lot more comprehensive.

**Comment by Adharainspace** [comment]
from what im seeing on here and the forums and discord  it seems like people dont really think this is a good idea, or at least, more people are vocally against this rather than for it. ill probably close this in a few hours when i get back to my computer, since i doubt the course'll change much

**Comment by MarkNais** [comment]
I think this is worth trying for a few rounds at least. I think many people are exaggerating their expected issues.

**Comment by Adharainspace** [comment]
i dont know how to test merge and idk, feels disrespectful to just go "oh you dont know what you really want" to what is a lot of people who have concerns. gonna eat dinner and then close this

**Pull request closed without merging**